### PR TITLE
Unwrap position fields and attributes to computedAttribute context

### PR DESCRIPTION
**Pull request by Abyss777**
Map all position fields and attributes to jexl context with help of reflection.

Expressions look more simple now.

Did not notice any changes in performance.

Added some tests to check if scripts can influence on the original position.

### CODE REVIEW

File: src/org/traccar/processing/ComputedAttributesHandler.java [hunk 1]
```diff
@@ -36,10 +39,32 @@ public ComputedAttributesHandler() {
         engine.setStrict(true);
     }
 
+    private MapContext prepareContext(Position position) {
+        MapContext result = new MapContext();
+        Method[] methods = position.getClass().getMethods();
+        for (Method method : methods) {
+            if (method.getName().startsWith("get") && method.getParameterTypes().length == 0
+                    && !method.getName().equals("getClass")) {
+                String name = method.getName().substring(3, 4).toLowerCase() + method.getName().substring(4);
```
Comment (tananaev) [inline src/org/traccar/processing/ComputedAttributesHandler.java]: You can do a small optimization here and use `Character.toLowerCase(...)` method on a single `char`.

File: src/org/traccar/processing/ComputedAttributesHandler.java [hunk 2]
```diff
@@ -36,10 +39,32 @@ public ComputedAttributesHandler() {
         engine.setStrict(true);
     }
 
+    private MapContext prepareContext(Position position) {
+        MapContext result = new MapContext();
+        Method[] methods = position.getClass().getMethods();
+        for (Method method : methods) {
+            if (method.getName().startsWith("get") && method.getParameterTypes().length == 0
+                    && !method.getName().equals("getClass")) {
```
Comment (tananaev) [inline src/org/traccar/processing/ComputedAttributesHandler.java]: I suggest some more generic solution for excluding methods here. Something like this:
```java
Set<Method> methods = new HashSet<>(Arrays.asList(position.getClass().getMethods()));
methods.removeAll(Arrays.asList(Object.class.getMethods()));
for (Method method : methods) {
    ...
```

**Comment by Abyss777** [comment]
Thank you for tips. Fixed.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**